percpu_macros/src/arch.rs

```diff
@@ -39,6 +39,15 @@ pub fn gen_offset(symbol: &Ident) -> proc_macro2::TokenStream {
 }
 
 pub fn gen_current_ptr(symbol: &Ident, ty: &Type) -> proc_macro2::TokenStream {
+    // For ARM architecture, we assume running in EL1
+    // and use `TPIDR_EL1` to store the base address of the per-CPU data area.
+    #[cfg(not(feature = "arm-el2"))]
+    let aarch64_tpidr = "TPIDR_EL1";
```

Comment:
Use `let aarch64_tpidr = if cfg!(feature = "arm_el2") {} else {}` can be more readable.

Reply:
done!